protocol/lavasession/consumer_session_manager.go

```diff
@@ -356,32 +394,71 @@ func (csm *ConsumerSessionManager) getValidProviderAddress(ignoredProvidersList
 	}
 
 	providers := csm.providerOptimizer.ChooseProvider(csm.validAddresses, ignoredProvidersList, cu, requestedBlock, OptimizerPerturbation)
```

Comment:
this can return an empty list without an error, we have to check there is at least one or err

Reply:
The logic of this method prevents calling ChooseProvider if it will return 0. Before calling we make sure that there are valid providers. 

```
	totalValidLength := validAddressesLength - ignoredProvidersListLength
	if totalValidLength <= 0 {
		utils.LavaFormatDebug("Pairing list empty", utils.Attribute{Key: "Provider list", Value: csm.validAddresses}, utils.Attribute{Key: "IgnoredProviderList", Value: ignoredProvidersList})
		err = PairingListEmptyError
		return
	}
```

But for the sake of certainty and the logic in the ChooseProvider (which can change), i agree we can add double check just to be sure:

Added: 3678366158ad49c457980c745d9fa2276d185bf5